packages/core/src/hooks/data/useApiUrl.spec.tsx

```diff
@@ -3,16 +3,110 @@ import { renderHook } from "@testing-library/react";
 import { MockJSONServer, TestWrapper } from "@test";
 
 import { useApiUrl } from "./";
+import * as UseResource from "../../hooks/resource/useResource";
 
 describe("useApiUrl Hook", () => {
-    it("returns context value", async () => {
-        const { result } = renderHook(() => useApiUrl(), {
-            wrapper: TestWrapper({
-                dataProvider: MockJSONServer,
-                resources: [{ name: "posts" }],
-            }),
+    describe.each(["meta", "options"])(`should be called with %s`, (key) => {
+        describe("when dataProvider is *not* provided", () => {
```

Comment:
```suggestion
        describe("when dataProvider is *not* provided", () => {
```
Can such a scenario happen?

Reply:
Here I mean `dataProviderName` param is undefined.